packages/xrpl/src/client/index.ts

```diff
@@ -661,48 +523,530 @@ class Client extends EventEmitter {
   }
 
   /**
+   * Autofills fields in a transaction. This will set `Sequence`, `Fee`,
+   * `lastLedgerSequence` according to the current state of the server this Client
+   * is connected to. It also converts all X-Addresses to classic addresses and
+   * flags interfaces into numbers.
+   *
    * @category Core
+   *
+   * @example
+   *
+   * ```ts
+   * const { Client } = require('xrpl')
+   *
+   * const client = new Client('wss://s.altnet.rippletest.net:51233')
+   *
+   * async function createAndAutofillTransaction() {
+   *   const transaction = {
+   *     TransactionType: 'Payment',
+   *     Account: 'rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh',
+   *     Destination: 'r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59',
+   *     Amount: '10000000' // 10 XRP in drops (1/1,000,000th of an XRP)
+   *   }
+   *
+   *   try {
+   *     const autofilledTransaction = await client.autofill(transaction)
+   *     console.log(autofilledTransaction)
+   *   } catch (error) {
+   *     console.error(`Failed to autofill transaction: ${error}`)
+   *   }
+   * }
+   *
+   * createAndAutofillTransaction()
+   * ```
+   *
+   * Autofill helps fill in fields which should be included in a transaction, but can be determined automatically
+   * such as `LastLedgerSequence` and `Fee`. If you override one of the fields `autofill` changes, your explicit
+   * values will be used instead. By default, this is done as part of `submit` and `submitAndWait` when you pass
+   * in an unsigned transaction along with your wallet to be submitted.
+   *
+   * @template T
+   * @param transaction - A {@link Transaction} in JSON format
+   * @param signersCount - The expected number of signers for this transaction.
+   * Only used for multisigned transactions.
+   * @returns The autofilled transaction.
    */
-  public autofill = autofill
+  public async autofill<T extends Transaction>(
```

Comment:
I really don't like the idea of moving all this code into the `Client` file - it makes it way more difficult to find the code that you need and it makes this file monstrously large

Reply:
Yeah Phu felt the same, but I don't think there's another solution. We could remove some of the body of the functions, but honestly that feels unnecessary to me and more confusing in my eyes. Open to a vote by reviewers for moving the body of the functions out to separate files as I did with `doFundWalletRequest`

Reply:
When you say it makes it more difficult to find the code that you need, what specifically are you referring to? (What was the easier path people had before which is gone?)

Reply:
Unsure if it's be much better - but if filesize is an issue, instead of `public autofill = autofill` we could just do a one-line wrapper here with the same docstring to avoid the "this" issue:

e.g. 
```
public async autofill<T extends Transaction> ( ... ) {
    return doAutofill(this, ...)
}
```

I think that'd keep the proper doc generation, although I could be wrong.

Reply:
You could go to the file in `src/sugar` that had the code you wanted to look at, instead of scrolling through an absolutely massive file (in general, large files are harder to work with and should be split up where possible)

Reply:
So does doc gen not work if you do something like this?
```js
/*
* doc stuff here
*/
public autofill = autofill
```

Reply:
> So does doc gen not work if you do something like this?
> 
> ```js
> /*
> * doc stuff here
> */
> public autofill = autofill
> ```

Correct

Reply:
> Unsure if it's be much better - but if filesize is an issue, instead of `public autofill = autofill` we could just do a one-line wrapper here with the same docstring to avoid the "this" issue:
> 
> e.g.
> 
> ```
> public async autofill<T extends Transaction> ( ... ) {
>     return doAutofill(this, ...)
> }
> ```
> 
> I think that'd keep the proper doc generation, although I could be wrong.

That's what I meant yup!

Reply:
Also you'd have to have a double docstring if you move it into a different file, since we require docstrings on file exports.

Reply:
> > Unsure if it's be much better - but if filesize is an issue, instead of `public autofill = autofill` we could just do a one-line wrapper here with the same docstring to avoid the "this" issue:
> > e.g.
> > ```
> > public async autofill<T extends Transaction> ( ... ) {
> >     return doAutofill(this, ...)
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I think that'd keep the proper doc generation, although I could be wrong.
> 
> That's what I meant yup!

Yeah, this would be better. I think it's fine to disable the docstrings (or have a dummy docstring)

Reply:
@justinr1234 Sorry for asking about what you already proposed - your replies didn't show up when I was writing the message.